### PR TITLE
Local request po cfor game config

### PR DESCRIPTION
**Pull request by charley43613**

### CODE REVIEW

File: gamelogic/socket_controllers/updatePreGame.js [hunk 1]
```diff
@@ -83,16 +86,111 @@ async function getPregameByRoomName(roomId){
...
+		//TODO replace with findOneandUpdate
+		// console.log("roomId PASSED ot find the fucking game config is " + JSON.stringify(roomId));
+		const tempPregame = await Pregame.findOne({roomId, "players.socketId": player });
+		console.log("temppregamE CONFIG previous socket assigned for coin decision was " + JSON.stringify(tempPregame.coinDecisionSocketId) + "while player socket is " + JSON.stringify(player) + "player coin decision WAS" + playerCoinDecision);;
+		//winningplayer will be re-assigned according to result
+		let coinFlipResult = flipCoin().toUpperCase();
+		let winningPlayer = player;
+		console.log('flip coin result is ' + coinFlipResult + ' while playerCo9inDecision was ' + playerCoinDecision);
+		if(!(coinFlipResult === playerCoinDecision)){
+			for(let i=0;i<tempPregame.players.length; i++){
```
[the remark this answers is not in the document]
Comment (charley43613) [reply gamelogic/socket_controllers/updatePreGame.js:106]: as long as findOne returns a map(i can't tell what its returning/what mongoose returns), but i aint worried about max 2 iteration loop in this case, i'll change to const for the vars

File: models/Game.js [hunk 2]
```diff
@@ -13,23 +13,29 @@ const GameSchema = new mongoose.Schema({
 			turn: false,
 			cards: [
 				{
+					//we could add 'inDeck' property, however we can deduce from the booleans below whether it is in the deck if it meets
+					//none of those categories, i.e. not isHand/isActive etc
+				
+					//user defined(us)
```
[the remark this answers is not in the document]
Comment (charley43613) [reply models/Game.js:19]: I'll try thi sout when i pass cards that are missing the properties, just gotta be able to add them at some point to the records is all

File: gamelogic/socket_controllers/updatePreGame.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jtmccutcheon) [reply gamelogic/socket_controllers/updatePreGame.js]: even if it doesnt .players is what you map and players will always be an array

File: models/Game.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jtmccutcheon) [reply models/Game.js:19]: the default will create it when it doesnt exist thats why you put it there

File: gamelogic/socket_controllers/updatePreGame.js [hunk 3]
```diff
@@ -41,58 +45,181 @@ async function updateGameConfig(roomId, player, deckIdPassed){
...
+		//TODO replace with findOneandUpdate
+		// console.log("roomId PASSED ot find the fucking game config is " + JSON.stringify(roomId));
+		const tempPregame = await Pregame.findOne({ roomId, "players.socketId": player });
+		console.log("temppregamE CONFIG previous socket assigned for coin decision was " + JSON.stringify(tempPregame.coinDecisionSocketId) + "while player socket is " + JSON.stringify(player) + "player coin decision WAS" + playerCoinDecision);;
+		//winningplayer will be re-assigned according to result
+		const coinFlipResult = flipCoin().toUpperCase();
+		const winningPlayer = player;
+		console.log('flip coin result is ' + coinFlipResult + ' while playerCo9inDecision was ' + playerCoinDecision);
+		if (!(coinFlipResult === playerCoinDecision)) {
+			for (let i = 0; i < tempPregame.players.length; i++) {
```
[the remark this answers is not in the document]
Comment (Jtmccutcheon) [reply gamelogic/socket_controllers/updatePreGame.js:108]: something along those lines

File: gamelogic/socket_controllers/updatePreGame.js [hunk 4]
```diff
@@ -41,58 +45,181 @@ async function updateGameConfig(roomId, player, deckIdPassed){
...
+
+		const newGame = new Game({
+			roomId,
+			players: [{ socketId: updatedPregame.players[0].socketId, turn: isPlayer1Winner, cards: updatedPregame.players[0].cards },
+			{ socketId: updatedPregame.players[1].socketId, turn: !isPlayer1Winner, cards: updatedPregame.players[1].cards },
+
+			]
+		});
+
+		const shit = await newGame.save();
```
Comment (Jtmccutcheon) [inline gamelogic/socket_controllers/updatePreGame.js:167]: this shouldnt need assignment you can just `await newGame.save();`
and the next variable can just be shit

File: gamelogic/socket_controllers/updatePreGame.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (charley43613) [reply gamelogic/socket_controllers/updatePreGame.js]: bear with me for now on this.

File: gamelogic/socket_controllers/updatePreGame.js (hunk 4 above)
Comment (charley43613) [reply gamelogic/socket_controllers/updatePreGame.js:171]: had only for loging purposes, can definitely remove if it bothers you

File: gamelogic/socket_controllers/updatePreGame.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (charley43613) [reply gamelogic/socket_controllers/updatePreGame.js]: There is no complexity difference in find(which finds first) and this. Whats the concern?

**Pull request merged**